validator/sawtooth_validator/execution/executor.py

```diff
@@ -77,47 +102,188 @@ def run(self):
-            if self._require_txn_processors and \
-                    header.family_name == 'sawtooth_config' and \
-                    processor_type not in self._processors:
-                # wait until required processor is registered:
-                LOGGER.info('Waiting for transaction processor (%s, %s, %s)',
-                            header.family_name,
-                            header.family_version,
-                            header.payload_encoding)
-
-                self._processors.wait_to_process(processor_type)
-
-            if processor_type not in self._processors:
-                raise Exception("internal error, no processor available")
-            processor = self._processors.get_next_of_type(processor_type)
+            config = self._config_view_factory.create_config_view(
+                txn_info.state_hash)
+            transaction_families = config.get_setting(
+                key=self._tp_config_key,
+                default_value="[]")
+            # After reading the transaction families required in configuration
+            # try to json.loads them into a python object
+            # If there is a misconfiguration, proceed as if there is no
+            # configuration.
+            try:
+                transaction_families = json.loads(transaction_families)
+                required_transaction_processors = [
+                    processor_iterator.ProcessorType(
+                        d.get('family'),
+                        d.get('version'),
+                        d.get('encoding')) for d in transaction_families]
+            except ValueError:
+                LOGGER.warning("sawtooth.validator.transaction_families "
+                               "misconfigured. Expecting a json array, found"
+                               " %s", transaction_families)
+                required_transaction_processors = []
+
+            config_processor = processor_iterator.ProcessorType(
+                name='sawtooth_config',
+                version='1.0',
+                encoding='application/protobuf')
+
+            # If there are required transaction processors, sawtooth_config
+            # should be one of them, so the TransactionExecutor doesn't
+            # automatically fail config transactions.
+            if len(required_transaction_processors) > 0 and config_processor \
+                    not in required_transaction_processors:
+                required_transaction_processors += [config_processor]
+
+            if processor_type not in required_transaction_processors:
+                # If there aren't any configured required transaction
+                # processors, wait for all transactions. This is useful
+                # for startup with the assumption being that the
+                # required transaction processors would be configured soon
+                # after startup.
+                if len(required_transaction_processors) == 0:
```

Comment:
If required_transaction_processors is of length 0, then that should be interpreted as "any".  This allows us to avoid explicit configuration of the sawtooth.validator.transaction_families setting at startup.